lib/ansible/modules/packaging/os/rhn_register.py

```diff
@@ -349,7 +351,13 @@ def main():
             systemorgid=dict(default=None, required=False),
             enable_eus=dict(default=False, type='bool'),
             channels=dict(default=[], type='list'),
-        )
+        ),
+        # username/password is required for state=absent, or if channels is not empty
```

Comment:
when `state` is `present`: either (`username` and `password`) or `activation_key` are required in [`register`](https://github.com/WhyIsThisOpen/ansible/blob/57cf604488ed3051fa111a288f008fe6c3444d06/lib/ansible/modules/packaging/os/rhn_register.py#L252) method. `register` method is called independently from `channels` value.

Reply:
FWIW, this was discussed on IRC and @abadger @samdoran and myself agreed that while the `required_if` and `required_together` is more compact and a mechanism meant to be leveraged by modules, given the complexity of the decision matrix required it ends up being simpler and more maintainable code without using those.